aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -23,7 +24,7 @@ namespace c10 {
  * so we can create a new cache instance when a kernel is looked up
  * from the dispatch table.
  */
-using KernelCacheCreatorFunction = std::unique_ptr<c10::KernelCache> ();
+using KernelCacheCreatorFunction = std::function<std::unique_ptr<c10::KernelCache> ()>;
```

Comment:
std::function? really?

Reply:
I'm open for better ideas. We need something that can store state though. I don't really like that, but it's the only way to support the legacy function pointer based API - the function pointer needs to be stored somewhere.

Reply:
The only other option would be to pass explicit `void* data` value like DataPtr, but I guess we already have std::function today so it's ok

Reply:
Note that it's only std::function on the cache creator, not on the actual kernel call. That means, we need to call a std::function whenever we look up a new kernel from the dispatch table, but then can call it multiple times without going through std::function.